### DISCUSSION TITLE
To consider the αcτµαlly pδrταblε εxεcµταblε (APE) format cross-platform executable

### DISCUSSION THREAD

**Discussion by alhabshi3k** [titzer/virgil#104] [Ideas]
Hi, I just saw your project Virgil, and the interesting part is the compilation of different targets (Native machine & Virtual Machine).
I don't know which language you used to produce the 1st compiler of Virgil, But I may guess it was C. The reason I am referring to this point is because of an open-source that tries to solve the issues of cross-compilation to native machine code from a different view. I will not give more details than the links to read, download, compile and test from the author (Justine Tunney) blogs posts.  Here are the links:
- [αcτµαlly pδrταblε εxεcµταblε](https://justine.lol/ape.html)
- [cosmopolitan libc](https://justine.lol/cosmopolitan/index.html) : "_your build-once run-anywhere c library_."

In summary, it is about compiling one-time to a native-machine code (At the moment is x86) and running on any operating system platform (Darwin, Linux, BSD, Window). I see the project is promising as yours.
